### PR TITLE
Add documentation for support of CP migration by extensions

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane-migration
/area documentation
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Adds documentation for support of control plane migration by extensions.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: docs/extensions/migration.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+* The `migrate` operation is used to ask the extension controller in the source seed to stop reconciling extension resources (in case they are requeued due to errors) and perform cleanup activities, if such are required. These cleanup activities might involve removing finalizers on resources in the shoot namespace that have been previously created by the extension controller and deleting them without actually deleting any resources external to the seed cluster.
+* The `restore` operation is used to ask extension controller in the destination seed to restore any state saved in the extension resource `status`, before performing the actual reconciliation.
+
+As with the [reconcile operation](https://github.com/gardener/gardener/blob/master/docs/extensions/reconcile-trigger.md), extension controllers must remove the `gardener.cloud/operation` annotation at the end of a successful reconciliation also when the current operation is `migrate` or `restore`.
```
Comment (plkokanov) [inline docs/extensions/migration.md]: Just a note here: It should be `unlike` the reconcile annotation as it is removed at the start of the operation, whereas the migrate and restore annotations are removed after a successful operation.

File: docs/extensions/migration.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      "version": 3,
+      "terraform_version": "0.11.14",
+      "serial": 2,
+      "lineage": "3a1e2faa-e7b6-f5f0-5043-368dd8ea6c10",
+      ...
+    }
+```
+
+Extension controllers that do not use a saved state and therefore do not require state migration could leave the `status.state` field as `nil` at the end of a successful reconciliation, and just perform a normal reconciliation when the current operation is `restore`.
+
```
Comment (plkokanov) [inline docs/extensions/migration.md:67]: Should we also add the `status.Resources` field which can be referred to by entries in the `status.State` field, for more clarity (sadly the infrastructure resource won't be appropriate for it)

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks, looks pretty good and informative, I have just a couple remarks.

File: docs/extensions/migration.md (hunk 2 above)
Comment (stoyanr) [reply docs/extensions/migration.md:67]: Done

File: docs/extensions/migration.md (hunk 1 above)
Comment (stoyanr) [reply docs/extensions/migration.md]: Done

**Review by plkokanov — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**